Extend settings component

jarolrod opened the pull request:
bitcoin-core/gui-qml#124 Introduced information.qml which is a mutation of `Setting.qml` but introduces an additional two loaders and respective properties to handle the scenarios where a setting doesn't use an option switch and needs to instead display a link, display an image, or take input.

The issue, despite large amounts of duplicate code is that, following this design pattern, we will be introducing a new mutation of `Settings.qml` everytime we have a scenario when there should be something aside from an option switch.

Instead of endless mutations, `Settings.qml` can just be made extensible by simply introducing a loader which gives it an `actionItem`. an `actionItem` is the interact-able element of this setting; that could be an option switch, a url, a text input box, or something completely new.

An example of something new, see the current simplification of the `AboutOptions.qml` developer settings button:

```diff
-    RowLayout {
-        Header {
-            Layout.fillWidth: true
-            center: false
-            header: qsTr("Developer options")
-            headerSize: 18
-            description: qsTr("Only use these if you have development experience")
-            descriptionSize: 15
-            descriptionMargin: 10
-            wrap: false
-            }
-        Loader {
-            Layout.fillWidth: true
-            Layout.preferredWidth: 0
-            Layout.alignment: Qt.AlignRight
-            Layout.rightMargin: 5
-            active: true
-            visible: active
-            sourceComponent: TextButton {
-                text: ">"
-                bold: false
-                rightalign: true
-                onClicked: {
-                    introductions.incrementCurrentIndex()
-                    swipeView.inSubPage = true
-                }
+    Setting {
+        Layout.fillWidth: true
+        header: qsTr("Developer options")
+        description: qsTr("Only use these if you have development experience")
+        actionItem: TextButton {
+            text: ">"
+            bold: false
+            rightalign: true
+            onClicked: {
+                introductions.incrementCurrentIndex()
+                swipeView.inSubPage = true
             }
```

[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/143)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/143)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/143)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/143)

shaavan commented on src/qml/components/ConnectionSettings.qml:
```diff
@@ -12,40 +12,48 @@ ColumnLayout {
         Layout.fillWidth: true
         header: qsTr("Proxy settings")
-        description: qsTr(">")
+        actionItem: ValueInput {
+            description: qsTr(">")
+        }
```
This should be converted to a TextButton instead of a ValueInput since this option symbolizes redirecting a new page.

jarolrod replied:
this will be updated to an icon so I'm not going to do this here

shaavan reviewed (commented):
Strong Concept ACK

- I agree with this change for all the reasons explained in the description. Especially because this helps remove redundant code while making the Settings.qml scalable.
- The code changes are complete and exhaustive and completely map the functionality with the master.
- While testing this PR on my laptop (Ubuntu 22.04, Qt version 5.15.3), I observed some key differences from the master.

## Testing differences:

1. Hovering over links no more shows the edit cursor.

**Screenshot:**

|Master|PR|
|---------|----|
|![Screenshot from 2022-08-03 18-43-03](https://user-images.githubusercontent.com/85434418/182646443-ceb269e6-d3d4-443d-a945-c2e37a4985b6.png)|![Screenshot from 2022-08-03 18-42-39](https://user-images.githubusercontent.com/85434418/182646354-75252e82-b3cc-49de-8362-6b7006328340.png)|

2. The optionswitch is misaligned from the divider line.

**Screenshot:**
|Master|PR|
|---------|----|
|![Screenshot from 2022-08-03 18-46-34](https://user-images.githubusercontent.com/85434418/182646673-011b45c8-cce5-43f8-9401-6bebcfd61bc2.png)|![Screenshot from 2022-08-03 18-46-07](https://user-images.githubusercontent.com/85434418/182646724-891f6851-2751-4fd6-b145-9f6e8e355eac.png)|

There is a few more change that needs to be fixed in master as well, which I would suggest adding to this PR. However, these can be taken up in separate PR as well.

1. The button for the Developers option on the About Options page is not right-aligned properly.

**Screenshot:**
![Screenshot from 2022-08-03 18-42-39](https://user-images.githubusercontent.com/85434418/182647904-81c59ea2-d6c0-4d4b-8695-2d19d442d96d.png)

jarolrod commented:
updated from ac4dbd2 to 0935349 ([PR143.01](https://github.com/jarolrod/gui-qml/tree/pr143.01) -> [PR143.02](https://github.com/jarolrod/gui-qml/tree/pr143.02), [diff](https://github.com/jarolrod/gui-qml/compare/pr143.01..pr143.02))

changes:
- rebase over changes on #141 
- update one temporary setting with no importance from an `ExternalLink` to `ValueInput` to prevent a warning in the terminal when running the gui

jarolrod commented:
updated from 0935349 to 3b53ae6 ([pr143.02](https://github.com/jarolrod/gui-qml/tree/pr143.02) -> [pr143.03](https://github.com/jarolrod/gui-qml/tree/pr143.03), [diff](https://github.com/jarolrod/gui-qml/compare/pr143.02..pr143.03))

changes: rebased over changes in #141

jarolrod commented:
updated from 3b53ae6 to 69d0778 ([pr143.03](https://github.com/jarolrod/gui-qml/tree/pr143.03) -> [pr143.04](https://github.com/jarolrod/gui-qml/tree/pr143r.04), [diff](https://github.com/jarolrod/gui-qml/compare/pr143.02..pr143r.04))

changes: rebased over changes in #141

hebasto reviewed (approved):
ACK 69d0778afcb9b8ad0b7cdeb8497abc365b9313ee, tested on Ubuntu 22.04.

hebasto commented:
@jarolrod 
> changes: rebased over changes in #141

One more rebase?

hebasto reviewed (approved):
re-ACK d4db4d155134ff249764ea1cd4df99468b8f7516

jarolrod commented:
updated from [d4db4d1 ](https://github.com/bitcoin-core/gui-qml/commit/d4db4d155134ff249764ea1cd4df99468b8f7516)to [ee82e9c ](https://github.com/bitcoin-core/gui-qml/commit/ee82e9c040a92a3324c4ad00502f974a258e737b)[Compare](https://github.com/bitcoin-core/gui-qml/compare/d4db4d155134ff249764ea1cd4df99468b8f7516..ee82e9c040a92a3324c4ad00502f974a258e737b)

changes: rebase over changes on master

hebasto reviewed (approved):
re-ACK ee82e9c040a92a3324c4ad00502f974a258e737b

Edited PR links in the PR description.

hebasto commented on src/qml/components/AboutOptions.qml:
```diff
@@ -9,56 +9,49 @@ import "../controls"
+        header: qsTr("Version")
+        actionItem: ExternalLink {
+            description: qsTr("v22.99.0-1e7564eca8a6 >")
+            link: "https://bitcoin.org/en/download"
```
Please remove this link or update it to https://bitcoincore.org/
